### PR TITLE
Hw 04

### PR DESCRIPTION
**Pull request by andreikasmykou**
I Deleted all files from earlier homework and Realized minimum requirements for HW_04, without DB and logger. May be in the next step.

### CODE REVIEW

File: src/middlewares/query-parser.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * @description
+ * Middleware for query parsing in request.
+ * Parse query string and populate `req.parsedQuery`
+ * with an object keyed and value by the query data.
+ *
+ * @param {Request} req
+ * @param {http.ServerResponse} res
+ * @param {Function} next
+ */
+export const queryParser = (req, res, next) => {
```
Comment (slowpokex) [inline src/middlewares/query-parser.js:12]: This solution is difficult. Please use https://nodejs.org/docs/latest/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost with true flag

File: src/middlewares/cookie-parser.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * Parse Cookie header and populate `req.parsedCookies`
+ * with an object keyed and value by the cookie data.
+ *
+ * @param {Request} req
+ * @param {http.ServerResponse} res
+ * @param {Function} next
+ */
+export const cookieParser = (req, res, next) => {
+  if (!req.headers.cookie) {
+    req.parsedCookies = 'Have no any cookies, sorry';
```
Comment (slowpokex) [inline src/middlewares/cookie-parser.js]: Just return empty object

File: src/http-servers/echo-server.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import http from 'http';
+
+const port = 3000;
+
+/**
+ * @description
+ * Simple echo server.
+ * Get request and pipe it to the response.
+ */
+http.createServer((req, res) => {
```
Comment (slowpokex) [inline src/http-servers/echo-server.js]: http.createServer((req, res) => req.pipe(res)) please try like here

File: src/app.js [hunk 4]
```diff
@@ -1,61 +1,80 @@
...
-  default :
-    console.log('You have not passed any actions'); // eslint-disable-line no-console
-    program.help();
-}
+  res.send(`Product with id = ${req.params.id} is not found.`);
+});
+
+//  /api/products/:id/reviews GET Return ALL reviews for a single product
+app.get('/api/products/:id/reviews', (req, res) => {
+  for (let key in products) {
```
Comment (slowpokex) [inline src/app.js]: please avoid for-in cycles. Use ForEach or for-of

File: src/app.js [hunk 5]
```diff
@@ -1,61 +1,80 @@
...
+    const rs = new Readable();
+    rs.push(JSON.stringify(productsObj, null, 2));
+    rs.push(null);
+    rs.pipe(fs.createWriteStream(`${__dirname}${path.sep}data${path.sep}products.json`));
+    res.send(product);
+  });
+
+//   /api/products/:id GET Return SINGLE product
+app.get('/api/products/:id', (req, res) => {
+  for (let key in products) {
```
Comment (slowpokex) [inline src/app.js]: Avoid for-in everywhere

**Review by slowpokex — CHANGES_REQUESTED** [review]

File: src/middlewares/query-parser.js (hunk 1 above)
Comment (andreikasmykou) [reply src/middlewares/query-parser.js:12]: Done)) I thought that we need to write it by ourselves.

File: src/middlewares/cookie-parser.js (hunk 2 above)
Comment (andreikasmykou) [reply src/middlewares/cookie-parser.js]: Done.

File: src/http-servers/echo-server.js (hunk 3 above)
Comment (andreikasmykou) [reply src/http-servers/echo-server.js]: Refactored

File: src/app.js (hunk 4 above)
Comment (andreikasmykou) [reply src/app.js]: Refactored with Object.keys and forEach.

File: src/app.js (hunk 5 above)
Comment (andreikasmykou) [reply src/app.js]: Same

**Review by slowpokex — APPROVED** [review]

**Pull request merged**